x/slashing/keeper/signing_info.go

```diff
@@ -1,94 +1,111 @@
 package keeper
 
 import (
+	"context"
 	"time"
 
+	"cosmossdk.io/errors"
 	storetypes "cosmossdk.io/store/types"
 	"github.com/bits-and-blooms/bitset"
-	"github.com/cockroachdb/errors"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/slashing/types"
 )
 
 // GetValidatorSigningInfo retruns the ValidatorSigningInfo for a specific validator
-// ConsAddress
-func (k Keeper) GetValidatorSigningInfo(ctx sdk.Context, address sdk.ConsAddress) (types.ValidatorSigningInfo, bool) {
-	store := ctx.KVStore(k.storeKey)
-
+// ConsAddress. If not found it returns ErrNoSigningInfoFound, but other errors
+// may be returned if there is an error reading from the store.
+func (k Keeper) GetValidatorSigningInfo(ctx context.Context, address sdk.ConsAddress) (types.ValidatorSigningInfo, error) {
+	store := k.storeService.OpenKVStore(ctx)
 	var info types.ValidatorSigningInfo
-	bz := store.Get(types.ValidatorSigningInfoKey(address))
+	bz, err := store.Get(types.ValidatorSigningInfoKey(address))
+	if err != nil {
+		return info, err
+	}
+
 	if bz == nil {
-		return info, false
+		return info, types.ErrNoSigningInfoFound
```

Comment:
This behavior change should be precise in the changelog imho.

Reply:
Added: `GetValidatorSigningInfo` now returns an error instead of a `found bool`, the error can be `nil` (found), `ErrNoSigningInfoFound` (not found) and any other error.